pkg/ruler/api_test.go

```diff
@@ -27,125 +27,219 @@ import (
-									State:  "inactive",
-									Health: "unknown",
-									Type:   "alerting",
-									Alerts: []*Alert{},
-								},
-							},
-							Interval: 60,
+		"should load and evaluate the configured rules": {
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      "group1",
+					Namespace: "namespace1",
+					User:      "user1",
+					Rules:     []*rulespb.RuleDesc{mockRecordingRuleDesc("UP_RULE", "up"), mockAlertingRuleDesc("UP_ALERT", "up < 1")},
+					Interval:  interval,
+				},
+			},
+			expectedRules: []*RuleGroup{
+				{
+					Name: "group1",
+					File: "namespace1",
+					Rules: []rule{
+						&recordingRule{
+							Name:   "UP_RULE",
+							Query:  "up",
+							Health: "unknown",
+							Type:   "recording",
+						},
+						&alertingRule{
+							Name:   "UP_ALERT",
+							Query:  "up < 1",
+							State:  "inactive",
+							Health: "unknown",
+							Type:   "alerting",
+							Alerts: []*Alert{},
 						},
 					},
+					Interval: 60,
 				},
 			},
 		},
-		"rules special characters": {
-			mockRules: mockSpecialCharRules,
-			userID:    "user1",
-			expectedResponse: response{
-				Status: "success",
-				Data: &RuleDiscovery{
-					RuleGroups: []*RuleGroup{
-						{
-							Name: ")(_+?/|group1+/?",
-							File: ")(_+?/|namespace1+/?",
-							Rules: []rule{
-								&recordingRule{
-									Name:   "UP_RULE",
-									Query:  "up",
-									Health: "unknown",
-									Type:   "recording",
-								},
-								&alertingRule{
-									Name:   "UP_ALERT",
-									Query:  "up < 1",
-									State:  "inactive",
-									Health: "unknown",
-									Type:   "alerting",
-									Alerts: []*Alert{},
-								},
-							},
-							Interval: 60,
+		"should load and evaluate only recording rules if alerting rules evaluation is disabled for the tenant": {
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      "group1",
+					Namespace: "namespace1",
+					User:      "user1",
+					Rules:     []*rulespb.RuleDesc{mockRecordingRuleDesc("UP_RULE", "up"), mockAlertingRuleDesc("UP_ALERT", "up < 1")},
+					Interval:  interval,
+				},
+			},
+			overrides: map[string]rulerLimits{
+				"user1": {
+					recordingRulesEvaluationEnabled: true,
+					alertingRulesEvaluationEnabled:  false,
+				},
+			},
+			expectedRules: []*RuleGroup{
+				{
+					Name: "group1",
+					File: "namespace1",
+					Rules: []rule{
+						&recordingRule{
+							Name:   "UP_RULE",
+							Query:  "up",
+							Health: "unknown",
+							Type:   "recording",
 						},
 					},
+					Interval: 60,
 				},
 			},
 		},
-		"federated rules": {
-			userID: "user1",
-			mockRules: map[string]rulespb.RuleGroupList{
+		"should load and evaluate only alerting rules if recording rules evaluation is disabled for the tenant": {
+			configuredRules: rulespb.RuleGroupList{
+				&rulespb.RuleGroupDesc{
+					Name:      "group1",
+					Namespace: "namespace1",
+					User:      "user1",
```

Comment:
```suggestion
					User:      userID,
```

Reply:
i think there a few other places with this, can you please inplace the constant there too